backend/src/main/kotlin/app/ehrenamtskarte/backend/freinet/webservice/FreinetApplicationMutationService.kt

```diff
@@ -0,0 +1,145 @@
+import app.ehrenamtskarte.backend.freinet.database.repos.FreinetAgencyRepository
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import graphql.schema.DataFetchingEnvironment
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.plugins.HttpRequestRetry
+import io.ktor.client.plugins.HttpResponseValidator
+import io.ktor.client.request.request
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.URLProtocol
+import io.ktor.http.path
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.slf4j.LoggerFactory
+
+@Suppress("unused")
+class FreinetApplicationMutationService {
+    private val logger = LoggerFactory.getLogger(FreinetApplicationMutationService::class.java)
+    private val httpClient = HttpClient {
+        install(HttpRequestRetry) {
+            retryOnServerErrors(maxRetries = 3)
+            retryOnException(maxRetries = 3, retryOnTimeout = false)
+            exponentialDelay()
+        }
+        HttpResponseValidator {
+            validateResponse { response ->
+                if (response.status != HttpStatusCode.OK) {
+                    throw Exception(response.status.toString())
+                }
+            }
+        }
+    }
+
+    private val jsonParser = Json { ignoreUnknownKeys = true }
+
+    @GraphQLDescription("Send application info to Freinet")
+    fun sendApplicationDataToFreinet(applicationId: Int, project: String, dfe: DataFetchingEnvironment): Boolean {
+        val context = dfe.graphQlContext.context
+        val admin = context.getAdministrator()
+        val projectConfig = dfe.graphQlContext.context.backendConfiguration.getProjectConfig(
+            project,
+        )
+        if (projectConfig.freinet == null) {
+            throw NotImplementedException()
+        }
+
+        return transaction {
+            val application = ApplicationRepository.findByIds(listOf(applicationId)).firstOrNull()
+                ?: throw NotFoundException("Application not found")
+
+            val regionId = application.regionId.value
+
+            if (!Authorizer.mayViewApplicationsInRegion(admin, regionId)) {
+                logger.warn("unauthorized access")
+                throw UnauthorizedException()
+            }
+
+            val freinetAgency = FreinetAgencyRepository.getFreinetAgencyByRegionId(regionId)
+                ?: throw FreinetAgencyNotFoundException()
+
+            if (!freinetAgency.dataTransferActivated) {
+                // Freinet is disabled for this region
+                throw FreinetDataTransferNotActivatedException()
```

Comment:
🔧 i wouldn't throw an error here, because every region that hasn't activated this in region settings (lots of them not using freinet) will recieve an error. just return false

Reply:
false = fail , true = success , if exception = 'FREINET_DATA_TRANSFER_NOT_ACTIVATED' it means ignore.

https://github.com/digitalfabrik/entitlementcard/blob/b86470a501258261394c2f84c85d936f8dcf150b/administration/src/bp-modules/cards/hooks/useCardGenerator.ts#L79-L84